aiida/cmdline/commands/graph.py

```diff
@@ -34,65 +31,57 @@ def __init__(self):
-        parser.add_argument('-d', '--descendant-depth', help="The maximum depth when "
-                "recursing through the descendants, if not set it will recurse to the end", type=PositiveInt)
-        parser.add_argument('--outputs', help="Always show all outputs of a calculation", action='store_true')
-        parser.add_argument('--inputs', help="Always show all inputs of a calculation", action='store_true')
-        parser.add_argument('-f', '--format', help="The output format, something that "
-            "can be recognized by graphvix (see http://www.graphviz.org/doc/info/output.html)", default='dot')
-        # Parse args and retrieve root pk
-        args = list(args)
-        parsed_args = parser.parse_args(args)
-        root_pk = parsed_args.ROOT
-
-        # Try to retrieve the node
-        try:
-            n = load_node(root_pk)
-        except NotExistent as e:
-            print >> sys.stderr, e.message
-            sys.exit(1)
-        exit_status, output_file_name = draw_graph(n, 
-                ancestor_depth=parsed_args.ancestor_depth, descendant_depth=parsed_args.descendant_depth, format=parsed_args.format,
-                include_calculation_inputs=parsed_args.inputs, include_calculation_outputs=parsed_args.outputs)
-        if not exit_status:
-            print "Output file is {}".format(output_file_name)
-
-
+    def cli(self, *args):
+        verdi()
+
+
+@verdi_graph.command('generate')
+@arguments.NODE('root_node')
+@click.option(
+    '-a',
+    '--ancestor-depth',
+    help="The maximum depth when recursing upwards, if not set it will recurse to the end",
+    type=click.IntRange(min=0))
+@click.option(
+    '-d',
+    '--descendant-depth',
+    help="The maximum depth when recursing through the descendants, if not set it will recurse to the end",
+    type=click.IntRange(min=0))
+@click.option(
+    '-o', 
+    '--outputs', 
+    is_flag=True, 
+    help="Always show all outputs of a calculation")
+@click.option(
+    '-i', 
+    '--inputs', 
+    is_flag=True, 
+    help="Always show all inputs of a calculation")
+@click.option(
+    '-f',
+    '--output-format',
+    help="The output format, something that can be recognized by graphvix"
+    "(see http://www.graphviz.org/doc/info/output.html)",
+    default='dot')
+@decorators.with_dbenv()
+def generate(root_node, ancestor_depth, descendant_depth, outputs, inputs, output_format):
+    """
+    Generate a graph given a ROOT_NODE user-specified by its pk.
+    """
+    from aiida.orm import load_node
+    from aiida.common.graph import draw_graph
+
+    exit_status, output_file_name = draw_graph(
+        root_node,
+        ancestor_depth=ancestor_depth,
+        descendant_depth=descendant_depth,
+        format=output_format,
+        include_calculation_inputs=inputs,
+        include_calculation_outputs=outputs)
+    if not exit_status:
+        echo.echo_success("Output file is {}".format(output_file_name))
```

Comment:
Maybe also include an `echo.echo_critical` if the file was not written?

Reply:
Done